src/smc-webapp/editor-html-md/editor-html-md.coffee

```diff
@@ -62,7 +62,9 @@ class exports.HTML_MD_Editor extends editor.FileEditor
             # canonicalize .wiki and .mediawiki (as used on github!) to "mediawiki"
             @ext = "mediawiki"
             @opts.mode = 'mediawiki'
-        else if @ext != 'java'
+        else if @ext == 'java'
+            $('.salvus-editor-textedit-buttonbar').hide()
```

Comment:
Hell no!   Do `@element.find('.salvus-editor-textedit-buttonbar').hide()`...

The above then this is scary, since you have two bugs that sort of cancel out, but if you understand what you just did, you'll see that this approach very badly breaks things, e.g,. if you have another .md file open in a tab...